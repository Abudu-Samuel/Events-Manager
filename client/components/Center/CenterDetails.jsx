import React from 'react';
import ReactPaginate from 'react-paginate';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import PropTypes from 'prop-types';
import { BounceLoader } from 'react-spinners';
import { Link } from 'react-router-dom';
import Navbar from '../common/Navbar';
import CenterInfo from '../Center/CenterInfo';
import * as userActions from '../../actions/actionCreator';

/**
 * @class CenterDetails
 *
 * @description CenterDetails component
 *
 * @param {object} event
 *
 * @extends {React.Component}
 */
export class CenterDetails extends React.Component {
  /**
   * Creates an instance of CenterDetails.
   *
   * @param {any} props
   *
   * @memberof CenterDetails
   */
  constructor(props) {
    super(props);
    this.state = {
      user: null,
      center: [],
      upcomingEventsData: {
        message: '',
        pagination: {},
        upcomingEvent: []
      },
      noEvent: false,
      loading: true,
      fetchingCenter: false
    };
    this.eventPaginate = this.eventPaginate.bind(this);
  }

  /**
   *@description react lifecycle
   *
   * @method componentDidMount
   *
   * @memberof CenterDetails
   *
   * @return {object} new afdfdfad
   */
  componentDidMount() {
    const { centerId } = this.props.match.params;
    this.props.singleCenter(centerId);
    this.props.slatedEvent(centerId, 1);
  }

  /**
   * @description react lifecycle
   *
   * @param {object} nextProps
   *
   * @return {object} updated stated
   *
   * @memberof CenterDetails
   */
  componentWillReceiveProps(nextProps) {
    if (nextProps.getSingleCenter) {
      this.setState({ center: nextProps.getSingleCenter, isAvailable: true, loading: false });
    } else {
      this.setState({ fetchingCenter: true });
    }
    if (nextProps.getSlatedEvents && nextProps.getSlatedEvents.upcomingEvent.length > 0) {
      this.setState({ upcomingEventsData: nextProps.getSlatedEvents, fetchingCenter: false, noEvent: false });
    } else {
      this.setState({ fetchingCenter: true, noEvent: true });
    }
  }

  /**
 * @description handle pagination for events
 *
 * @returns {object}  paginated page
 *
 * @param {any} pageData
 *
 * @memberof CenterDetails
 */
  eventPaginate(pageData) {
    const { centerId } = this.props.match.params;
    const nextEventPage = pageData.selected + 1;
    this.props.slatedEvent(centerId, nextEventPage);
  }

  /**
   *@method render
   *
   * @returns {jsx} Jsx respresentation of the dom
   *
   * @memberof CenterDetails
   */
  render() {
    const { center, upcomingEventsData, noEvent } = this.state;
    return (
      <div className="space">
        <Navbar />
        <BounceLoader
          color={'#123abc'}
          loading={this.state.loading}
        />
        <CenterInfo
          center={center}
          upcomingEventsData={upcomingEventsData}
          noEvent={noEvent}
        />{
          noEvent ? <div className="space text-center">
            <i className="fa fa-exclamation-triangle no-event" />
            <h5 className="font-weight-bold mb-5">No event allocated to this center,
              Click <strong><Link to={`/center/${center.id}/addevent`} className="add-event">Here</Link></strong> to book a center
            </h5>
          </div> :
            <ReactPaginate
              previousLabel="Previous"
              nextLabel="Next"
              breakLabel={<a href="">...</a>}
              breakClassName="page-link"
              onPageChange={this.eventPaginate}
              pageCount={Number(this.props.eventPage)}
              containerClassName="pagination pagination-lg custom-pagination"
              pageLinkClassName="page-link"
              nextLinkClassName="page-link"
              previousLinkClassName="page-link"
              disabledClassName="disabled"
              pageClassName="page-item"
              previousClassName="page-item"
              nextClassName="page-item"
              activeClassName="active"
              subContainerClassName="pages pagination"
            />
        }
      </div>
    );
  }
}

CenterDetails.PropTypes = {
  centerId: PropTypes.number,

};

/**
 * @description Redux connect parameter - mapStateToProps
 *
 * @param {function} state
 *
 * @return {object} mapped state
 */
const mapStateToProps = state => ({
  getSingleCenter: state.centers.center,
  getSlatedEvents: state.events.upcomingEventsData,
  eventPage: state.events.eventPage
});

/**
 * @description Redux connect parameter - mapDispatchToProps
 *
 * @param {function} dispatch
 *
 * @return {object} mapped dispatch
 */
export const mapDispatchToProps = dispatch => bindActionCreators({
  singleCenter: centerData => userActions.singleCenter(centerData),
  slatedEvent: (eventId, page) => userActions.slatedEvent(eventId, page)
}, dispatch);


export default connect(mapStateToProps, mapDispatchToProps)(CenterDetails);

