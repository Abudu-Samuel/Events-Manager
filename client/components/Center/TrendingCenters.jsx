import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

const TrendingCenter = ({
  centers,
  getCenterId,
}) => (
  <div className="container">
    <section id="popular centers">
      <h2 className="mb-3 slot font-weight-bold text-center">Latest Centers</h2>
      <div className="row mb-4">
        {
          centers.center.map(center => (<div className="col-md-4 mb-4"
            key={center.id}>
            <div className="card hoverable text-center">
              <img className="img-fluid hoverable max" src={center.image}
                alt="Card image cap" />
              <div className="card-body">
                <h4 className="card-title">{center.name}</h4>
                <p className="card-text">
                  {
                    center.description.split('').length > 40 ?
                      `${center.description.slice(0, 40)}...` :
                      center.description
                  }
                </p>
                <button
                  id="viewCenter"
                  className="btn btn-mycolor btn-sm"
                  data-centerid={center.id}
                  onClick={getCenterId}>
                  <Link to={`/centers/${center.id}`}>Details</Link>
                </button>
              </div>
            </div>
          </div>))
        }
      </div>
    </section>
  </div>
);

TrendingCenter.propTypes = {
  centers: PropTypes.object,
  getCenterId: PropTypes.func
};

export default TrendingCenter;
