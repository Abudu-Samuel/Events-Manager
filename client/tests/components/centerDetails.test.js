import React from 'react';
import { shallow } from 'enzyme';
import expect from 'expect';
import { CenterDetails, mapDispatchToProps } from '../../components/Center/CenterDetails';
import centerMockedData from '../__mocks__/centerMockedData';

const props = {
  singleCenter: () => Promise.resolve(),
  slatedEvent: () => Promise.resolve(),
  match: {
    params: {
      eventId: 1
    }
  }
};

const centerPage = {
  nextCenterPage: {
    pageData: {
      selected: 1
    }
  }
};

describe('<CenterDetails />', () => {
  it('should render EventDetails component correctly correctly', () => {
    const wrapper = shallow(<CenterDetails {...props}/>);
    wrapper.setProps({
      getSlatedEvents: {
        center: centerMockedData.centerData,
        upcomingEvent: {
          length: 1
        }
      }
    });
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.state().fetchingCenter).toBe(false);
  });

  it('should render EventDetails component correctly correctly', () => {
    const wrapper = shallow(<CenterDetails {...props}/>);
    wrapper.setProps({
      getSingleCenter: {
        center: centerMockedData.centerData
      }
    });
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.state().isAvailable).toBe(true);
  });

  it('should render CenterDetails component correctly correctly', () => {
    const wrapper = shallow(<CenterDetails {...props}/>);
    wrapper.setProps({
      center: null
    });
    expect(wrapper).toMatchSnapshot();
    wrapper.instance().eventPaginate(centerPage);
    expect(wrapper.state().fetchingCenter).toBe(true);
  });
});

describe('the mapDispatchToProps function', () => {
  it('should return an object of functions to be passed as props', () => {
    const dispatchFn = jest.fn();
    const propsObj = mapDispatchToProps(dispatchFn);

    expect(propsObj).toEqual({
      singleCenter: expect.any(Function),
      slatedEvent: expect.any(Function),
    });

    propsObj.singleCenter();
    propsObj.slatedEvent();

    expect(dispatchFn).toHaveBeenCalledTimes(2);
  });
});
