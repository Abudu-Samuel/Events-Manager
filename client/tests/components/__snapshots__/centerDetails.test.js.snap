// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CenterDetails /> should render CenterDetails component correctly correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CenterDetails
    center={null}
    match={
      Object {
        "params": Object {
          "eventId": 1,
        },
      }
    }
    singleCenter={[Function]}
    slatedEvent={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Connect(Navbar) />,
        <onlyUpdateForKeys(Loader)
          color="#123abc"
          loading={true}
          size={60}
          sizeUnit="px"
        />,
        <CenterInfo
          center={Array []}
          noEvent={true}
          upcomingEventsData={
            Object {
              "message": "",
              "pagination": Object {},
              "upcomingEvent": Array [],
            }
          }
        />,
        <div
          className="space text-center"
        >
          <i
            className="fa fa-exclamation-triangle no-event"
          />
          <h5
            className="font-weight-bold mb-5"
          >
            No event allocated to this center, Click 
            <strong>
              <Link
                className="add-event"
                replace={false}
                to="/center/undefined/addevent"
              >
                Here
              </Link>
            </strong>
             to book a center
          </h5>
        </div>,
      ],
      "className": "space",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "color": "#123abc",
          "loading": true,
          "size": 60,
          "sizeUnit": "px",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "center": Array [],
          "noEvent": true,
          "upcomingEventsData": Object {
            "message": "",
            "pagination": Object {},
            "upcomingEvent": Array [],
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <i
              className="fa fa-exclamation-triangle no-event"
            />,
            <h5
              className="font-weight-bold mb-5"
            >
              No event allocated to this center, Click 
              <strong>
                <Link
                  className="add-event"
                  replace={false}
                  to="/center/undefined/addevent"
                >
                  Here
                </Link>
              </strong>
               to book a center
            </h5>,
          ],
          "className": "space text-center",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "fa fa-exclamation-triangle no-event",
            },
            "ref": null,
            "rendered": null,
            "type": "i",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "No event allocated to this center, Click ",
                <strong>
                  <Link
                    className="add-event"
                    replace={false}
                    to="/center/undefined/addevent"
                  >
                    Here
                  </Link>
                </strong>,
                " to book a center",
              ],
              "className": "font-weight-bold mb-5",
            },
            "ref": null,
            "rendered": Array [
              "No event allocated to this center, Click ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Link
                    className="add-event"
                    replace={false}
                    to="/center/undefined/addevent"
                  >
                    Here
                  </Link>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Here",
                    "className": "add-event",
                    "replace": false,
                    "to": "/center/undefined/addevent",
                  },
                  "ref": null,
                  "rendered": "Here",
                  "type": [Function],
                },
                "type": "strong",
              },
              " to book a center",
            ],
            "type": "h5",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Connect(Navbar) />,
          <onlyUpdateForKeys(Loader)
            color="#123abc"
            loading={true}
            size={60}
            sizeUnit="px"
          />,
          <CenterInfo
            center={Array []}
            noEvent={true}
            upcomingEventsData={
              Object {
                "message": "",
                "pagination": Object {},
                "upcomingEvent": Array [],
              }
            }
          />,
          <div
            className="space text-center"
          >
            <i
              className="fa fa-exclamation-triangle no-event"
            />
            <h5
              className="font-weight-bold mb-5"
            >
              No event allocated to this center, Click 
              <strong>
                <Link
                  className="add-event"
                  replace={false}
                  to="/center/undefined/addevent"
                >
                  Here
                </Link>
              </strong>
               to book a center
            </h5>
          </div>,
        ],
        "className": "space",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "color": "#123abc",
            "loading": true,
            "size": 60,
            "sizeUnit": "px",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "center": Array [],
            "noEvent": true,
            "upcomingEventsData": Object {
              "message": "",
              "pagination": Object {},
              "upcomingEvent": Array [],
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <i
                className="fa fa-exclamation-triangle no-event"
              />,
              <h5
                className="font-weight-bold mb-5"
              >
                No event allocated to this center, Click 
                <strong>
                  <Link
                    className="add-event"
                    replace={false}
                    to="/center/undefined/addevent"
                  >
                    Here
                  </Link>
                </strong>
                 to book a center
              </h5>,
            ],
            "className": "space text-center",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "fa fa-exclamation-triangle no-event",
              },
              "ref": null,
              "rendered": null,
              "type": "i",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "No event allocated to this center, Click ",
                  <strong>
                    <Link
                      className="add-event"
                      replace={false}
                      to="/center/undefined/addevent"
                    >
                      Here
                    </Link>
                  </strong>,
                  " to book a center",
                ],
                "className": "font-weight-bold mb-5",
              },
              "ref": null,
              "rendered": Array [
                "No event allocated to this center, Click ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <Link
                      className="add-event"
                      replace={false}
                      to="/center/undefined/addevent"
                    >
                      Here
                    </Link>,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "Here",
                      "className": "add-event",
                      "replace": false,
                      "to": "/center/undefined/addevent",
                    },
                    "ref": null,
                    "rendered": "Here",
                    "type": [Function],
                  },
                  "type": "strong",
                },
                " to book a center",
              ],
              "type": "h5",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<CenterDetails /> should render EventDetails component correctly correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CenterDetails
    getSlatedEvents={
      Object {
        "center": Object {
          "capacity": 123,
          "description": "Awesome center",
          "image": "testing.img",
          "isAvailable": true,
          "location": "yaba",
          "name": "test center",
          "price": 123456,
          "state": "lagos",
        },
        "upcomingEvent": Object {
          "length": 1,
        },
      }
    }
    match={
      Object {
        "params": Object {
          "eventId": 1,
        },
      }
    }
    singleCenter={[Function]}
    slatedEvent={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Connect(Navbar) />,
        <onlyUpdateForKeys(Loader)
          color="#123abc"
          loading={true}
          size={60}
          sizeUnit="px"
        />,
        <CenterInfo
          center={Array []}
          noEvent={false}
          upcomingEventsData={
            Object {
              "center": Object {
                "capacity": 123,
                "description": "Awesome center",
                "image": "testing.img",
                "isAvailable": true,
                "location": "yaba",
                "name": "test center",
                "price": 123456,
                "state": "lagos",
              },
              "upcomingEvent": Object {
                "length": 1,
              },
            }
          }
        />,
        <PaginationBoxView
          activeClassName="active"
          breakClassName="page-link"
          breakLabel={
            <a
              href=""
            >
              ...
            </a>
          }
          containerClassName="pagination pagination-lg custom-pagination"
          disableInitialCallback={false}
          disabledClassName="disabled"
          marginPagesDisplayed={3}
          nextClassName="page-item"
          nextLabel="Next"
          nextLinkClassName="page-link"
          onPageChange={[Function]}
          pageClassName="page-item"
          pageCount={NaN}
          pageLinkClassName="page-link"
          pageRangeDisplayed={2}
          previousClassName="page-item"
          previousLabel="Previous"
          previousLinkClassName="page-link"
          subContainerClassName="pages pagination"
        />,
      ],
      "className": "space",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "color": "#123abc",
          "loading": true,
          "size": 60,
          "sizeUnit": "px",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "center": Array [],
          "noEvent": false,
          "upcomingEventsData": Object {
            "center": Object {
              "capacity": 123,
              "description": "Awesome center",
              "image": "testing.img",
              "isAvailable": true,
              "location": "yaba",
              "name": "test center",
              "price": 123456,
              "state": "lagos",
            },
            "upcomingEvent": Object {
              "length": 1,
            },
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "activeClassName": "active",
          "breakClassName": "page-link",
          "breakLabel": <a
            href=""
          >
            ...
          </a>,
          "containerClassName": "pagination pagination-lg custom-pagination",
          "disableInitialCallback": false,
          "disabledClassName": "disabled",
          "marginPagesDisplayed": 3,
          "nextClassName": "page-item",
          "nextLabel": "Next",
          "nextLinkClassName": "page-link",
          "onPageChange": [Function],
          "pageClassName": "page-item",
          "pageCount": NaN,
          "pageLinkClassName": "page-link",
          "pageRangeDisplayed": 2,
          "previousClassName": "page-item",
          "previousLabel": "Previous",
          "previousLinkClassName": "page-link",
          "subContainerClassName": "pages pagination",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Connect(Navbar) />,
          <onlyUpdateForKeys(Loader)
            color="#123abc"
            loading={true}
            size={60}
            sizeUnit="px"
          />,
          <CenterInfo
            center={Array []}
            noEvent={false}
            upcomingEventsData={
              Object {
                "center": Object {
                  "capacity": 123,
                  "description": "Awesome center",
                  "image": "testing.img",
                  "isAvailable": true,
                  "location": "yaba",
                  "name": "test center",
                  "price": 123456,
                  "state": "lagos",
                },
                "upcomingEvent": Object {
                  "length": 1,
                },
              }
            }
          />,
          <PaginationBoxView
            activeClassName="active"
            breakClassName="page-link"
            breakLabel={
              <a
                href=""
              >
                ...
              </a>
            }
            containerClassName="pagination pagination-lg custom-pagination"
            disableInitialCallback={false}
            disabledClassName="disabled"
            marginPagesDisplayed={3}
            nextClassName="page-item"
            nextLabel="Next"
            nextLinkClassName="page-link"
            onPageChange={[Function]}
            pageClassName="page-item"
            pageCount={NaN}
            pageLinkClassName="page-link"
            pageRangeDisplayed={2}
            previousClassName="page-item"
            previousLabel="Previous"
            previousLinkClassName="page-link"
            subContainerClassName="pages pagination"
          />,
        ],
        "className": "space",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "color": "#123abc",
            "loading": true,
            "size": 60,
            "sizeUnit": "px",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "center": Array [],
            "noEvent": false,
            "upcomingEventsData": Object {
              "center": Object {
                "capacity": 123,
                "description": "Awesome center",
                "image": "testing.img",
                "isAvailable": true,
                "location": "yaba",
                "name": "test center",
                "price": 123456,
                "state": "lagos",
              },
              "upcomingEvent": Object {
                "length": 1,
              },
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "activeClassName": "active",
            "breakClassName": "page-link",
            "breakLabel": <a
              href=""
            >
              ...
            </a>,
            "containerClassName": "pagination pagination-lg custom-pagination",
            "disableInitialCallback": false,
            "disabledClassName": "disabled",
            "marginPagesDisplayed": 3,
            "nextClassName": "page-item",
            "nextLabel": "Next",
            "nextLinkClassName": "page-link",
            "onPageChange": [Function],
            "pageClassName": "page-item",
            "pageCount": NaN,
            "pageLinkClassName": "page-link",
            "pageRangeDisplayed": 2,
            "previousClassName": "page-item",
            "previousLabel": "Previous",
            "previousLinkClassName": "page-link",
            "subContainerClassName": "pages pagination",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<CenterDetails /> should render EventDetails component correctly correctly 2`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CenterDetails
    getSingleCenter={
      Object {
        "center": Object {
          "capacity": 123,
          "description": "Awesome center",
          "image": "testing.img",
          "isAvailable": true,
          "location": "yaba",
          "name": "test center",
          "price": 123456,
          "state": "lagos",
        },
      }
    }
    match={
      Object {
        "params": Object {
          "eventId": 1,
        },
      }
    }
    singleCenter={[Function]}
    slatedEvent={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Connect(Navbar) />,
        <onlyUpdateForKeys(Loader)
          color="#123abc"
          loading={false}
          size={60}
          sizeUnit="px"
        />,
        <CenterInfo
          center={
            Object {
              "center": Object {
                "capacity": 123,
                "description": "Awesome center",
                "image": "testing.img",
                "isAvailable": true,
                "location": "yaba",
                "name": "test center",
                "price": 123456,
                "state": "lagos",
              },
            }
          }
          noEvent={true}
          upcomingEventsData={
            Object {
              "message": "",
              "pagination": Object {},
              "upcomingEvent": Array [],
            }
          }
        />,
        <div
          className="space text-center"
        >
          <i
            className="fa fa-exclamation-triangle no-event"
          />
          <h5
            className="font-weight-bold mb-5"
          >
            No event allocated to this center, Click 
            <strong>
              <Link
                className="add-event"
                replace={false}
                to="/center/undefined/addevent"
              >
                Here
              </Link>
            </strong>
             to book a center
          </h5>
        </div>,
      ],
      "className": "space",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "color": "#123abc",
          "loading": false,
          "size": 60,
          "sizeUnit": "px",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "center": Object {
            "center": Object {
              "capacity": 123,
              "description": "Awesome center",
              "image": "testing.img",
              "isAvailable": true,
              "location": "yaba",
              "name": "test center",
              "price": 123456,
              "state": "lagos",
            },
          },
          "noEvent": true,
          "upcomingEventsData": Object {
            "message": "",
            "pagination": Object {},
            "upcomingEvent": Array [],
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <i
              className="fa fa-exclamation-triangle no-event"
            />,
            <h5
              className="font-weight-bold mb-5"
            >
              No event allocated to this center, Click 
              <strong>
                <Link
                  className="add-event"
                  replace={false}
                  to="/center/undefined/addevent"
                >
                  Here
                </Link>
              </strong>
               to book a center
            </h5>,
          ],
          "className": "space text-center",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "fa fa-exclamation-triangle no-event",
            },
            "ref": null,
            "rendered": null,
            "type": "i",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "No event allocated to this center, Click ",
                <strong>
                  <Link
                    className="add-event"
                    replace={false}
                    to="/center/undefined/addevent"
                  >
                    Here
                  </Link>
                </strong>,
                " to book a center",
              ],
              "className": "font-weight-bold mb-5",
            },
            "ref": null,
            "rendered": Array [
              "No event allocated to this center, Click ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Link
                    className="add-event"
                    replace={false}
                    to="/center/undefined/addevent"
                  >
                    Here
                  </Link>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Here",
                    "className": "add-event",
                    "replace": false,
                    "to": "/center/undefined/addevent",
                  },
                  "ref": null,
                  "rendered": "Here",
                  "type": [Function],
                },
                "type": "strong",
              },
              " to book a center",
            ],
            "type": "h5",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Connect(Navbar) />,
          <onlyUpdateForKeys(Loader)
            color="#123abc"
            loading={false}
            size={60}
            sizeUnit="px"
          />,
          <CenterInfo
            center={
              Object {
                "center": Object {
                  "capacity": 123,
                  "description": "Awesome center",
                  "image": "testing.img",
                  "isAvailable": true,
                  "location": "yaba",
                  "name": "test center",
                  "price": 123456,
                  "state": "lagos",
                },
              }
            }
            noEvent={true}
            upcomingEventsData={
              Object {
                "message": "",
                "pagination": Object {},
                "upcomingEvent": Array [],
              }
            }
          />,
          <div
            className="space text-center"
          >
            <i
              className="fa fa-exclamation-triangle no-event"
            />
            <h5
              className="font-weight-bold mb-5"
            >
              No event allocated to this center, Click 
              <strong>
                <Link
                  className="add-event"
                  replace={false}
                  to="/center/undefined/addevent"
                >
                  Here
                </Link>
              </strong>
               to book a center
            </h5>
          </div>,
        ],
        "className": "space",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "color": "#123abc",
            "loading": false,
            "size": 60,
            "sizeUnit": "px",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "center": Object {
              "center": Object {
                "capacity": 123,
                "description": "Awesome center",
                "image": "testing.img",
                "isAvailable": true,
                "location": "yaba",
                "name": "test center",
                "price": 123456,
                "state": "lagos",
              },
            },
            "noEvent": true,
            "upcomingEventsData": Object {
              "message": "",
              "pagination": Object {},
              "upcomingEvent": Array [],
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <i
                className="fa fa-exclamation-triangle no-event"
              />,
              <h5
                className="font-weight-bold mb-5"
              >
                No event allocated to this center, Click 
                <strong>
                  <Link
                    className="add-event"
                    replace={false}
                    to="/center/undefined/addevent"
                  >
                    Here
                  </Link>
                </strong>
                 to book a center
              </h5>,
            ],
            "className": "space text-center",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "fa fa-exclamation-triangle no-event",
              },
              "ref": null,
              "rendered": null,
              "type": "i",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "No event allocated to this center, Click ",
                  <strong>
                    <Link
                      className="add-event"
                      replace={false}
                      to="/center/undefined/addevent"
                    >
                      Here
                    </Link>
                  </strong>,
                  " to book a center",
                ],
                "className": "font-weight-bold mb-5",
              },
              "ref": null,
              "rendered": Array [
                "No event allocated to this center, Click ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <Link
                      className="add-event"
                      replace={false}
                      to="/center/undefined/addevent"
                    >
                      Here
                    </Link>,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "Here",
                      "className": "add-event",
                      "replace": false,
                      "to": "/center/undefined/addevent",
                    },
                    "ref": null,
                    "rendered": "Here",
                    "type": [Function],
                  },
                  "type": "strong",
                },
                " to book a center",
              ],
              "type": "h5",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
